codegen/src/test/java/software/amazon/awssdk/codegen/customization/processors/CustomSdkShapesProcessorTest.java

```diff
@@ -0,0 +1,160 @@
+        listShape.setType("list");
+
+        Member member = new Member();
+        String memberName = "Event";
+        member.setShape(memberName);
+        listShape.setMember(member);
+
+        String shapeName = "ListShape";
+        shapes.put(shapeName, listShape);
+        customSdkShapes.setShapes(shapes);
+        config.setCustomSdkShapes(customSdkShapes);
+
+        new IntermediateModelBuilder(C2jModels.builder()
+                                              .serviceModel(serviceModel)
+                                              .customizationConfig(config)
+                                              .build())
+            .build();
+
+        assertThat(serviceModel.getShapes().containsKey(shapeName)).isTrue();
+        assertThat(serviceModel.getShapes().containsValue(listShape)).isTrue();
+        assertThat(serviceModel.getShape(shapeName).getType().equals("list"));
+        assertThat(serviceModel.getShape(shapeName).getMembers().containsKey(memberName));
+
+    }
+
+    @Test
+    public void setCustomCustomizedShape_isAddedToModel() {
+        CustomSdkShapes customSdkShapes = config.getCustomSdkShapes();
+        Map<String, Shape> shapes = new HashMap<>();
+        shapes.putAll(customSdkShapes.getShapes());
+
+        Shape stringShape = new Shape();
+        stringShape.setType("string");
+
+        String shapeName = "StringShape";
+        shapes.put(shapeName, stringShape);
+
+        Shape customizedShape = new Shape();
+        customizedShape.setType(shapeName);
+        String customizedShapeName = "CustomizedShape";
+        shapes.put(customizedShapeName, customizedShape);
+
+
+        customSdkShapes.setShapes(shapes);
+        config.setCustomSdkShapes(customSdkShapes);
+
+        new IntermediateModelBuilder(C2jModels.builder()
+                                              .serviceModel(serviceModel)
+                                              .customizationConfig(config)
+                                              .build())
+            .build();
+
+        assertThat(serviceModel.getShapes().containsKey(customizedShapeName)).isTrue();
+        assertThat(serviceModel.getShapes().containsValue(customizedShape)).isTrue();
+        assertThat(serviceModel.getShape(customizedShapeName).getType().equals(shapeName));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void setCustomNullShape_throwsNullPointerException() {
+        CustomSdkShapes customSdkShapes = config.getCustomSdkShapes();
+        Map<String, Shape> shapes = new HashMap<>();
+        shapes.putAll(customSdkShapes.getShapes());
+
+        Shape stringShape = new Shape();
+        stringShape.setType(null);
+
+        String shapeName = "NullShape";
+        shapes.put(shapeName, stringShape);
+        customSdkShapes.setShapes(shapes);
+        config.setCustomSdkShapes(customSdkShapes);
+
+        // Throws NullPointerException
+        new IntermediateModelBuilder(C2jModels.builder()
```

Comment:
```
        assertThatNullPointerException()
            .isThrownBy(() ->  MethodToBetested())
            .withMessage("XYZ must not be null.");
```

Reply:
updated, without withMessage since no message is returned in the exception